### PR TITLE
Add docs for `HTMLElement.writingSuggestions`

### LINKED ISSUE #502: Add support for writingsuggestions attribute
### Proposal

The WHATWG html spec now supports a new attribute called writingsuggestions to control UA provided writing assistance. An MDN attribute should be written to document this attribute.

### Browser support

Not yet.

### Tasks

_No response_

### Dependencies

_No response_

### Additional information

_No response_

### Are you willing to support this work?

_No response_

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Additional details

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/writingsuggestions
https://html.spec.whatwg.org/multipage/interaction.html#dom-writingsuggestions

### Related issues and pull requests

Fixes https://github.com/mdn/mdn/issues/502
Relates to https://github.com/mdn/content/pull/35445
Relates to https://github.com/whatwg/html/pull/10018

### CODE REVIEW

**Review by estelle — COMMENTED** [review]
the interface property is boolean, returning only true or false, not enumerated.

**Comment by skyclouds2001** [comment]
> the interface property is boolean, returning only true or false, not enumerated.

no, this property is a enumerated property, not a boolean property

image

File: files/en-us/web/api/htmlelement/writingsuggestions/index.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+{{APIRef("HTML DOM")}}
+
+The **`writingSuggestions`** property of the {{domxref("HTMLElement")}} interface is a string indicating if browser-provided writing suggestions should be enabled under the scope of the element or not.
+
+It reflects the value of the [`writingsuggestions`](/en-US/docs/Web/HTML/Global_attributes/writingsuggestions) HTML global attribute.
+
+## Value
+
+An enumerated value; possible values are:
```
[the remark this answers is not in the document]
Comment (skyclouds2001) [reply files/en-us/web/api/htmlelement/writingsuggestions/index.md:17]: > In my tests, if the empty attribute is set, the property value is `true`, as a boolean not a string.

my test on chrome shows it is a enumerated property

File: files/en-us/web/api/htmlelement/writingsuggestions/index.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Examples
+
+The following example shows how to disable writing suggestions offered by user agents via script:
+
+```js
+const element = document.querySelector("input");
+
+// writing suggestions offered by user agents will be disabled by script manually on this input
+element.writingSuggestions = "manual";
```
[the remark this answers is not in the document]
Comment (skyclouds2001) [reply files/en-us/web/api/htmlelement/writingsuggestions/index.md:32]:
```suggestion
element.writingSuggestions = "true";
```

File: files/en-us/web/api/htmlelement/index.md [hunk 3]
```diff
@@ -86,6 +86,8 @@ _Also inherits properties from its parent, {{DOMxRef("Element")}}._
   - : A boolean value representing the translation.
 - {{DOMxRef("HTMLElement.virtualKeyboardPolicy")}} {{Experimental_Inline}}
   - : A string indicating the on-screen virtual keyboard behavior on devices such as tablets, mobile phones, or other devices where a hardware keyboard may not be available, if the element's content is editable (for example, it is an {{htmlelement("input")}} or {{htmlelement("textarea")}} element, or an element with the [`contenteditable`](/en-US/docs/Web/HTML/Global_attributes/contenteditable) attribute set).
+- {{DOMxRef("HTMLElement.writingSuggestions")}}
+  - : A string indicating if browser-provided writing suggestions should be enabled under the scope of the element or not.
```
[the remark this answers is not in the document]
Comment (skyclouds2001) [reply files/en-us/web/api/htmlelement/index.md:90]: it is a enumerated property

File: files/en-us/web/api/htmlelement/writingsuggestions/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (skyclouds2001) [reply files/en-us/web/api/htmlelement/writingsuggestions/index.md:17]: but it is true that the property only return `true` or `false`, no the empty string

File: files/en-us/web/api/htmlelement/writingsuggestions/index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+The **`writingSuggestions`** property of the {{domxref("HTMLElement")}} interface is a string indicating if browser-provided writing suggestions should be enabled under the scope of the element or not.
+
+It reflects the value of the [`writingsuggestions`](/en-US/docs/Web/HTML/Global_attributes/writingsuggestions) HTML global attribute.
+
+## Value
+
+An enumerated value; possible values are:
+
+- `"true"` or an empty string (`""`)
```
Comment (skyclouds2001) [inline files/en-us/web/api/htmlelement/writingsuggestions/index.md:19]:
```suggestion
- `"true"`
```

File: files/en-us/web/api/htmlelement/writingsuggestions/index.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+## Examples
+
+The following example shows how to disable writing suggestions offered by user agents via script:
+
+```js
+const element = document.querySelector("input");
+
+// writing suggestions offered by user agents will be disabled by script manually on this input
+element.writingSuggestions = "true";
```
Comment (skyclouds2001) [inline files/en-us/web/api/htmlelement/writingsuggestions/index.md:32]:
```suggestion
element.writingSuggestions = "false";
```

File: files/en-us/web/api/htmlelement/writingsuggestions/index.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  - : The browser does not automatically show the virtual keyboard: showing/hiding the virtual keyboard is handled manually by the script.
+
+## Examples
+
+The following example shows how to disable writing suggestions offered by user agents via script:
+
+```js
+const element = document.querySelector("input");
+
+// writing suggestions offered by user agents will be disabled by script manually on this input
```
Comment (estelle) [inline files/en-us/web/api/htmlelement/writingsuggestions/index.md:31]:
```suggestion
// disable user agent writing suggestions offered
```

**Review by estelle — APPROVED** [review]
I stand corrected. Thank you.

Thanks for the PR! 🎉

File: files/en-us/web/api/htmlelement/writingsuggestions/index.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  - : The browser does not automatically show the virtual keyboard: showing/hiding the virtual keyboard is handled manually by the script.
+
+## Examples
+
+The following example shows how to disable writing suggestions offered by user agents via script:
+
+```js
+const element = document.querySelector("input");
+
+// disable user agent writing suggestions offered
```
Comment (estelle) [inline files/en-us/web/api/htmlelement/writingsuggestions/index.md:31]:
```suggestion
// disable user agent writing suggestions
```

**Comment by littlemanguyer-cmyk** [comment]
You could just about put my name in to font bust it

**Comment by littlemanguyer-cmyk** [comment]
OK

**Pull request merged**